src/controls/mdfbChips.js

```diff
@@ -28,16 +28,32 @@ module.exports = function ($timeout) {
       if (scope.field.transformFunc) {
         var varToWatch = 'form.' + scope.field.name + '.$viewValue'
         scope.$watch(varToWatch, function (value, oldValue) {
+          var promises = []
           for (var i = 0; i < value.length; i++) {
             if (!value[i].text) {
               var code = angular.copy(value[i].code)
-              value[i].text = getSetOptionResult(scope.field.name, i, code)
+              value[i].text = getSetOptionResult(scope.field.name, i, code, promises)
```

Comment:
`getSetOptionResult` doesn't actually return anything so i'm not sure if this assignment is necessary. A better way to do this would be to have the `getSetOptionResult` return a promise rather than accepting all the promises and adding to it.

Reply:
Unless there is something I'm not understanding happening here, which is possible.